src/layers/base-layer.js

```diff
@@ -873,7 +873,7 @@ class Layer {
     const dataUpdateTriggers = this.getDataUpdateTriggers(layerDataset);
     const triggerChanged = this.getChangedTriggers(dataUpdateTriggers);
 
-    if (triggerChanged.getMeta) {
+    if (triggerChanged.getMeta || triggerChanged.getData) {
```

Comment:
for this case, you should change `getDataUpdateTriggers`  in base-layer, and add `allData` to the return,

```js
 return {
    getData: {datasetId: id, allData, columns, filteredIndex},
    getMeta: {datasetId: id, allData, columns},
...
}
```

Reply:
sounds good, will do

Reply:
@heshan0131 updated as requested